client/cypress/integration/query/parameter_spec.js

```diff
@@ -0,0 +1,99 @@
+describe('Parameter', () => {
+  beforeEach(() => {
+    cy.login();
+    cy.visit('/queries/new');
+
+    cy.getByTestId('QueryEditor')
+      .get('.ace_text-input')
+      .type("SELECT '{{}{{}test-parameter}}' AS parameter{esc}", { force: true });
+  });
+
+  afterEach(() => {
+    cy.getByTestId('SaveButton').click();
+    cy.url().should('match', /\/queries\/\d+\/source/);
+  });
```

Comment:
You sure this is needed? Can leave new query unsaved.

Reply:
It's either save the query or [this](https://github.com/getredash/redash/pull/3748/commits/c3d521fb9cc6a787d4d7791fe696204daacb2bb6) workaround :grimacing:

Reply:
Oh right right!